src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/StartToolStep.groovy

```diff
@@ -283,34 +242,42 @@ class StartToolStep extends Step {
 
             Process process = new ProcessBuilder().command(args.toCommandArray()).start()
 
-            boolean isStarted = false
-            long endTimeMillis = System.currentTimeMillis() + (long) timeout * 1000L
-            while (timeout <= 0 || System.currentTimeMillis() < endTimeMillis) {
-                if ((isStarted = process.isAlive())) {
-                    break
-                } else {
-                    Thread.sleep(1000)
+            boolean  isConnected = checkToolConnection()
+            int exitCode = 0
+            if (!isConnected) {
+                try  {
+                    exitCode = process.exitValue()
+                }  catch (IllegalThreadStateException ignore){
+                    throw new AbortException(
+                            "Timeout of ${this.timeout} seconds exceeded for connecting to ${toolName}! " +
+                                    "Please ensure the tool is correctly configured and consider restarting it.")
                 }
             }
+            if (exitCode == 0) {
+                return
+            }
 
-            if (!isStarted) {
-                throw new AbortException(
-                        "Timeout of ${this.timeout} seconds exceeded for starting ${toolName}! " +
-                        "Please ensure that the tool is correctly configured and accessible.")
+            if (exitCode == 99) {
+                throw new AbortException("No valid license found for ${toolName}! " +
+                        "Please ensure the license is not expired or corrupted.")
             }
+
+            throw new AbortException(
+                    "Timeout of ${this.timeout} seconds exceeded for connecting to ${toolName}! " +
+                            "Please ensure the tool is correctly configured and consider restarting it.")
```

Comment:
Having the same Exception for two different behaviors is odd. Here we might want to inform the user (or better us as support) that ecu.test did not start correctly and exited during within the timeout. Adding the exit code to the message.